ci/ci/github.py

```diff
@@ -354,12 +355,22 @@ def _hail_github_status_from_statuses(statuses_json):
             self.review_state = review_state
             self.target_branch.state_changed = True
 
+    def _update_intended_github_status_from_build_state(self):
+        intended_github_status = self.github_status_from_build_state()
+        if intended_github_status != self.intended_github_status:
+            self.intended_github_status = intended_github_status
+            self.target_branch.state_changed = True
+
     async def _start_build(self, dbpool, batch_client):
         assert await self.authorized(dbpool)
 
         # clear current batch
         self.batch = None
         self.build_state = None
+        intended_github_status = self.github_status_from_build_state()
+        if intended_github_status != self.intended_github_status:
+            self.intended_github_status = intended_github_status
+            self.target_branch.state_changed = True
```

Comment:
I'd suggest a `set_build_state` and put this inside.

Reply:
build_state is already a "property" whose setter is non-standard, but this seems a bit too sneaky to me. I'll add `set_build_state` rather than further overloading `build_state =`